api/include/opentelemetry/nostd/span.h

```diff
@@ -1,14 +1,31 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 #pragma once
 
-#include <array>
-#include <cassert>
-#include <cstddef>
-#include <exception>
-#include <iterator>
-#include <type_traits>
+#ifdef HAVE_CPP_STDLIB
+#  include "opentelemetry/std/span.h"
+#else
```

Comment:
Should there be check for`HAVE_ABSEIL` macro too, and include `absl/span.h` if set.

Reply:
I was unconditionally using either `std::span` or `gsl::span` since GSL span implementation is guarantee to be 100% conformant, whereas `absl::Span` is not guarnateed to be conformant. I added a brief description to the markdown document.... Basically for now I'd use variant from `absl::variant` , because there is no other option that would compile with Visual Studio 2015. But not using absl for Span. I can see if we can use that in a separate PR.

Reply:
@lalitb - this is described in the markdown doc here:
https://github.com/open-telemetry/opentelemetry-cpp/blob/maxgolov/nostd_stl/docs/building-with-stdlib.md#note-on-gslspan-vs-abslspan